### PR TITLE
run_structure_prediction.py accepts comma separated list of input folds and optionally dedicated output_directories for each fold

### PR DESCRIPTION
**Pull request by maurerv**

### CODE REVIEW

File: alphapulldown/scripts/run_structure_prediction.py [hunk 1]
```diff
@@ -224,8 +222,17 @@ def main(argv):
     data = create_custom_info(parsed_input)
     all_interactors = create_interactors(data, FLAGS.features_directory)
     objects_to_model = [] 
-    for interactors in all_interactors:
-        object_to_model, flags_dict, postprocess_flags, output_dir = pre_modelling_setup(interactors, FLAGS)
+
+    if len(FLAGS.input) != len(FLAGS.output_directory):
+        FLAGS.output_directory *= len(FLAGS.input)
```
[the remark this answers is not in the document]
Comment (maurerv) [reply alphapulldown/scripts/run_structure_prediction.py:227]: Exactly!

**Comment by maurerv** [comment]
@dingquanyu from your example PR at https://github.com/KosinskiLab/AlphaPulldownSnakemake/pull/13 it seemed like you wanted run_multimer_jobs.py to use a single output directory and create subdirectories for each fold according to use_ap_style. 

We could extend run_multimer_jobs.py to allow multiple output_paths, but since run_multimer_jobs.py uses the file-based fold specification, where the user might not know the number of folds beforehand, I think having a single output directory makes the most sense

**Comment by dingquanyu** [comment]
> @dingquanyu from your PR at [KosinskiLab/AlphaPulldownSnakemake#13](https://github.com/KosinskiLab/AlphaPulldownSnakemake/pull/13) it seemed like you wanted run_multimer_jobs.py to use a single output directory and create subdirectories for each fold according to use_ap_style.
> 
> We could extend run_multimer_jobs.py to allow multiple output_paths, but since run_multimer_jobs.py uses the file-based fold specification, where the user might not know the number of folds beforehand, I think having a single output directory makes the most sense

I see. This means in the snakemake pipeline, you will bypass run_multimer_jobs.py and launch run_structure_prediction.py directly with a cluster of jobs ?

**Comment by maurerv** [comment]
Exactly. I added a checkpoint that performs the clustering and then extended the current rule using run_structure_prediction.py to run on each cluster separately. This way we don't need additional rules

**Comment by dingquanyu** [comment]
> Exactly. I added a checkpoint that performs the clustering and then extended the current rule using run_structure_prediction.py to run on each cluster separately. This way we don't need additional rules.
> 
> I just pushed these changes for reference [bfa71c7ac5d013a0c1aea3b78fc347381a3ca06c](https://github.com/KosinskiLab/AlphaPulldownSnakemake/commit/bfa71c7ac5d013a0c1aea3b78fc347381a3ca06c)

I see. Thanks for the commit. Now it makes sense to me.

**Review by dingquanyu — APPROVED** [review]

**Pull request merged**